File: CustomActionSheet-OC/CustomActionSheet/ViewController.m
//
//  ViewController.m
//  CustomActionSheet
//
//  Created by yaoshuai on 2016/12/24.
//  Copyright © 2016年 ys. All rights reserved.
//

#import "ViewController.h"
#import "LCActionSheet.h"

@interface ViewController ()<LCActionSheetDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)defaultAction:(id)sender {
    LCActionSheet *actionSheet = [LCActionSheet sheetWithTitle:@"title"
                                                      delegate:self
                                             cancelButtonTitle:@"取消"
                                             otherButtonTitles:@"分享", @"删除", @"确认", nil];
    [actionSheet show];
}

- (IBAction)customStyleAction:(id)sender {
    LCActionSheet *actionSheet     = [[LCActionSheet alloc] initWithTitle:nil
                                                                 delegate:self
                                                        cancelButtonTitle:@"Cancel"
                                                        otherButtonTitles:@"Button 1", @"Button 2", @"Button 3", @"Button 4", @"Button 5", nil];
    actionSheet.title              = @"This is a very very very very very very very very very very very very very very very very very very very very very very very very very very very long title~";
    actionSheet.cancelButtonTitle  = @"Close";
    [actionSheet appendButtonTitles:@"Button 6", @"Button 7", nil];
    actionSheet.titleColor         = [UIColor orangeColor];
    actionSheet.buttonColor        = [UIColor greenColor];
    actionSheet.titleFont          = [UIFont boldSystemFontOfSize:15.0f];
    actionSheet.buttonFont         = [UIFont boldSystemFontOfSize:15.0f];
    actionSheet.buttonHeight       = 60.0f;
    actionSheet.scrolling          = YES;
    actionSheet.visibleButtonCount = 3.6f;
    actionSheet.darkViewNoTaped    = YES;
    actionSheet.unBlur             = YES;
    
    // V 2.1.0+ use `destructiveButtonIndexSet` instead `redButtonIndexSet`.
    actionSheet.destructiveButtonIndexSet = [NSSet setWithObjects:@0, @2, nil];
    actionSheet.destructiveButtonColor    = [UIColor blueColor];
    
    // V 2.7.0+
    actionSheet.titleEdgeInsets = UIEdgeInsetsMake(10, 20, 30, 40);
    
    [actionSheet show];
}

- (IBAction)blockAction:(id)sender {
    LCActionSheet *actionSheet = [LCActionSheet sheetWithTitle:@"Block LCActionSheet" cancelButtonTitle:@"Cancel" clicked:^(LCActionSheet *actionSheet, NSInteger buttonIndex) {
        
        NSLog(@"clickedButtonAtIndex: %d", (int)buttonIndex);
        
    } otherButtonTitles:@"Button 1", @"Button 2", @"Button 3", @"Button 4", @"Button 5", @"Button 6", nil];
    
    actionSheet.blurEffectStyle = UIBlurEffectStyleLight;
    
    actionSheet.scrolling          = YES;
    actionSheet.visibleButtonCount = 3.6f;
    
    actionSheet.willPresentHandle = ^(LCActionSheet *actionSheet) {
        NSLog(@"willPresentActionSheet");
    };
    
    actionSheet.didPresentHandle = ^(LCActionSheet *actionSheet) {
        NSLog(@"didPresentActionSheet");
    };
    
    actionSheet.willDismissHandle = ^(LCActionSheet *actionSheet, NSInteger buttonIndex) {
        NSLog(@"willDismissWithButtonIndex: %d", (int)buttonIndex);
    };
    
    actionSheet.didDismissHandle = ^(LCActionSheet *actionSheet, NSInteger buttonIndex) {
        NSLog(@"didDismissWithButtonIndex: %d", (int)buttonIndex);
    };
    
    [actionSheet show];
}



#pragma mark - LCActionSheetDelegate

- (void)actionSheet:(LCActionSheet *)actionSheet clickedButtonAtIndex:(NSInteger)buttonIndex {
    NSLog(@"clickedButtonAtIndex: %d", (int)buttonIndex);
}

- (void)willPresentActionSheet:(LCActionSheet *)actionSheet {
    NSLog(@"willPresentActionSheet");
}

- (void)didPresentActionSheet:(LCActionSheet *)actionSheet {
    NSLog(@"didPresentActionSheet");
}

- (void)actionSheet:(LCActionSheet *)actionSheet willDismissWithButtonIndex:(NSInteger)buttonIndex {
    NSLog(@"willDismissWithButtonIndex: %d", (int)buttonIndex);
}

- (void)actionSheet:(LCActionSheet *)actionSheet didDismissWithButtonIndex:(NSInteger)buttonIndex {
    NSLog(@"didDismissWithButtonIndex: %d", (int)buttonIndex);
}

@end

File: CustomActionSheet-Swift/CustomActionSheet/ViewController.swift
//
//  ViewController.swift
//  CustomActionSheet
//
//  Created by yaoshuai on 2016/12/24.
//  Copyright © 2016年 ys. All rights reserved.
//

import UIKit

class ViewController: UIViewController,LCActionSheetDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    // 方式一：使用代理方法执行按钮事件
    @IBAction func defaultActionSheetButtonClick(_ sender: UIButton) {
        
        // title可以为nil
        // 取消按钮：Index = 0
        // 其他按钮：按照数组中的顺序 1、2、3……
        let actionSheet = LCActionSheet(title: "this is the title", delegate: self, cancelButtonTitle: "取消", otherButtonTitleArray: ["分享","确认","删除"])
        
        actionSheet?.show()
    }
    
    // 自定义样式
    @IBAction func customActionSheetButtonClick(_ sender: UIButton) {
        
        let actionSheet = LCActionSheet(title: nil, delegate: self, cancelButtonTitle: "取消", otherButtonTitleArray: ["分享","确认","删除"])
        
        actionSheet?.title = "This is a very very very very very very very very very very very very very very very very very very very very very very very very very very very long title"
        actionSheet?.titleFont = UIFont.boldSystemFont(ofSize: 15)
        actionSheet?.titleColor = UIColor.orange
        actionSheet?.titleEdgeInsets = UIEdgeInsetsMake(10, 20, 30, 40);
        
        actionSheet?.buttonColor = UIColor.green
        actionSheet?.buttonFont = UIFont.boldSystemFont(ofSize: 15)
        actionSheet?.buttonHeight = 60.0;
        
        actionSheet?.canScrolling = true;
        actionSheet?.visibleButtonCount = 3.6;
        actionSheet?.darkViewNoTaped = true;
        actionSheet?.unBlur = true;
        actionSheet?.destructiveButtonColor = UIColor.blue
        
        actionSheet?.show()
    }
    
    // 方式三：使用闭包方式执行按钮事件
    @IBAction func blockActionSheetButtonClick(_ sender: UIButton) {
        
        // title可以为nil
        // 取消按钮：Index = 0
        // 其他按钮：按照数组中的顺序 1、2、3……
        let actionSheet = LCActionSheet(title: "this is the title", cancelButtonTitle: "取消", clicked: { (actionSheet, buttonIndex) in
            print("cancelButtonAtIndex:\(buttonIndex)")
        }, otherButtonTitleArray: ["按钮一","按钮二","按钮三","按钮四","按钮五","按钮六"])
        
        actionSheet?.blurEffectStyle = UIBlurEffectStyle.light
        actionSheet?.canScrolling = true
        actionSheet?.visibleButtonCount = 3.6
        
        actionSheet?.willPresentHandle = { (actionSheet) -> () in
            print("willPresentActionSheet")
        }
        actionSheet?.didPresentHandle = { (actionSheet) in
            print("didPresentActionSheet")
        }
        actionSheet?.willDismissHandle = { (actionSheet,buttonIndex:Int) in
            print("willDismissWithButtonIndex:\(buttonIndex)")
        }
        actionSheet?.didDismissHandle = { (actionSheet,buttonIndex:Int) in
            print("didDismissWithButtonIndex:\(buttonIndex)")
        }
        
        actionSheet?.show()
    }
    
    // MARK: - LCActionSheetDelegate
    func actionSheet(_ actionSheet: LCActionSheet!, clickedButtonAt buttonIndex: Int) {
        print("clickedButtonAtIndex: \(buttonIndex)")
    }
    
    func willPresent(_ actionSheet: LCActionSheet!) {
        print("willPresentActionSheet")
    }
    func didPresent(_ actionSheet: LCActionSheet!) {
        print("didPresentActionSheet")
    }
    
    func actionSheet(_ actionSheet: LCActionSheet!, willDismissWithButtonIndex buttonIndex: Int) {
        print("willDismissWithButtonIndex：\(buttonIndex)")
    }
    func actionSheet(_ actionSheet: LCActionSheet!, didDismissWithButtonIndex buttonIndex: Int) {
        print("didDismissWithButtonIndex：\(buttonIndex)")
    }
}

